api/api_ethereum.go

```diff
@@ -917,3 +1166,164 @@ func (api *EthereumAPI) rpcMarshalHeader(head *types.Header) (map[string]interfa
 
 	return result, nil
 }
+
+func EthDoCall(ctx context.Context, b Backend, args EthTransactionArgs, blockNrOrHash rpc.BlockNumberOrHash, overrides *EthStateOverride, timeout time.Duration, globalGasCap uint64) ([]byte, uint64, error) {
+	defer func(start time.Time) { logger.Debug("Executing EVM call finished", "runtime", time.Since(start)) }(time.Now())
+
+	st, header, err := b.StateAndHeaderByNumberOrHash(ctx, blockNrOrHash)
+	if st == nil || err != nil {
+		return nil, 0, err
+	}
+	if err := overrides.Apply(st); err != nil {
+		return nil, 0, err
+	}
+
+	// Setup context so it may be cancelled the call has completed
+	// or, in case of unmetered gas, setup a context with a timeout.
+	var cancel context.CancelFunc
+	if timeout > 0 {
+		ctx, cancel = context.WithTimeout(ctx, timeout)
+	} else {
+		ctx, cancel = context.WithCancel(ctx)
+	}
+	// Make sure the context is cancelled when the call has completed
+	// this makes sure resources are cleaned up.
+	defer cancel()
+
+	// TODO-Klaytn: Klaytn is using fixed baseFee as now.
+	fixedBaseFee := new(big.Int).SetUint64(params.BaseFee)
+	intrinsicGas, err := types.IntrinsicGas(args.data(), args.To == nil, b.ChainConfig().Rules(header.Number))
+	if err != nil {
+		return nil, 0, err
+	}
+	msg, err := args.ToMessage(globalGasCap, fixedBaseFee, intrinsicGas)
+	if err != nil {
+		return nil, 0, err
+	}
+	evm, vmError, err := b.GetEVM(ctx, msg, st, header, vm.Config{})
```

Comment:
Um.. This is just question.
There is `NoBaseFee` in vm.Config in Ethereum code.
Don't we need that?

Reply:
That option is used in [preCheck](https://github.com/ethereum/go-ethereum/blob/fb3a6528cfa49f623570575c4fe9e8a716cfcdf7/core/state_transition.go#L238-L256) to determine whether or not to do a sanity check about London related values.

I'd like to share the scope of my PR.
* I tried not to change data structures related to the StateTransition part as much as possible. 
* It because that is core part of Klaytn and need some lots of change related with it.
* For this, I used the GasPrice field only. 
  * Even if user specify MaxPriorityPerGas and MaxFeePerGas, GasPrice field is calculated and used based on that  values.